### PR TITLE
framework/ble_manager: apply whitelist APIs

### PR DESCRIPTION
**Pull request by an4967**
- apply whitelist APIs
- add whitelist usage example

### CODE REVIEW

File: framework/include/ble_manager/ble_scan.h [hunk 1]
```diff
@@ -62,17 +62,30 @@ typedef struct {
 	void(*ble_client_device_scanned_cb)(ble_scanned_device* scanned_device);
 } ble_scan_callback_list;
 
+/* This is a heuristic value. It can be customized following a system evn */
```
Comment (deepaksrma) [inline framework/include/ble_manager/ble_scan.h:65]: i think, is it env not evn?

File: framework/include/ble_manager/ble_scan.h [hunk 2]
```diff
@@ -62,17 +62,30 @@ typedef struct {
...
  * Description:
  *   Start BLE adv scanning
  *
  * Input Parameters:
- *   filter    - If this value is NULL, start to scan for BLE adv till 
- *               'ble_client_stop_scan' is called.
+ *   filter    - If this value is NULL, all filters are disabled.
  *               To get specific BLE adv info, filter value should be filled
  *               with exact value.
+ *               If scan_ducation is zero, the scan timer will be set for SCAN_MAX_TIMEOUT.
```
Comment (deepaksrma) [inline framework/include/ble_manager/ble_scan.h:88]: *duration

File: framework/src/ble_manager/ble_manager_state.c [hunk 3]
```diff
@@ -408,6 +403,26 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
...
+		int i;
+		uint16_t count = 0;
+		blemgr_msg_params *param = (blemgr_msg_params *)msg->param;
+		ble_addr *addr = (ble_addr *)param->param[0];
+		uint16_t size = *(uint16_t *)param->param[1];
+		for (i = 0; i < SCAN_WHITELIST_SIZE; i++) {
+			if (g_scan_whitelist[i].use == SCAN_WHITELIST_IN_USE) {
+				memcpy(&addr[count++], &g_scan_whitelist[i].addr, sizeof(ble_addr));
+			}
+			if (count >= size) {
```
Comment (sangam-swami) [inline framework/src/ble_manager/ble_manager_state.c:418]: do we need to check ">" also here? I guess size minimum will be > 0 ?

File: apps/examples/ble_rmc/ble_rmc_main.c [hunk 4]
```diff
@@ -472,37 +472,111 @@ int ble_rmc_main(int argc, char *argv[])
...
+			}
+			RMC_LOG(RMC_CLIENT_TAG, "Clear whitelist Success\n");
+		} else if (argc == 3 && strncmp(argv[2], "list", 5) == 0) {
+			ble_addr addr_list[10] = { 0, };
+			ble_addr *addr;
+			uint16_t count = 0;
+			int i;
+			count = ble_scan_whitelist_list(addr_list, 10);
+
+			RMC_LOG(RMC_CLIENT_TAG, "Total List : %d\n", count);
```
Comment (jeongchanKim) [inline apps/examples/ble_rmc/ble_rmc_main.c:530]: `ble_scan_whitelist_list` returns the unsigned value, so I think it is better to print the `count` with `%u`.

File: apps/examples/ble_rmc/ble_rmc_main.c [hunk 5]
```diff
@@ -472,37 +472,111 @@ int ble_rmc_main(int argc, char *argv[])
...
-				return 0;
+				goto ble_rmc_done;
+			}
+		} else if (argc >= 3 && argv[2][0] == '2') {
+			RMC_LOG(RMC_CLIENT_TAG, "Scan Start with WhiteList!\n");
+
+			uint32_t scan_time = 0;
+			if (argc == 4) {
+				scan_time = atoi(argv[3]);
+				RMC_LOG(RMC_CLIENT_TAG, "Timer : %d s\n", scan_time);
```
Comment (jeongchanKim) [inline apps/examples/ble_rmc/ble_rmc_main.c:559]: `scan_time` is unsigned type variable.

File: apps/examples/ble_rmc/ble_rmc_main.c [hunk 6]
```diff
@@ -472,37 +472,111 @@ int ble_rmc_main(int argc, char *argv[])
...
 			}
-		} else if (argc == 3 && argv[2][0] == '2') {
-			printf("Start with filter!\n");
 
 			ble_scan_filter filter = { 0, };
-			memcpy(&(filter.raw_data), ble_filter, sizeof(ble_filter));
-			filter.raw_data_length = sizeof(ble_filter);
-			filter.scan_duration = 1000;
+			filter.raw_data_length = 0;
+			filter.scan_duration = scan_time * 1000;
```
[the remark this answers is not in the document]
Comment (an4967) [reply apps/examples/ble_rmc/ble_rmc_main.c:564]: I will comment about this in the code.

File: framework/src/ble_manager/ble_manager_state.c (hunk 3 above)
Comment (an4967) [reply framework/src/ble_manager/ble_manager_state.c:418]: My purpose was copying addr list to the array as it allowed.
I will change this.

File: apps/examples/ble_rmc/ble_rmc_main.c (hunk 4 above)
Comment (an4967) [reply apps/examples/ble_rmc/ble_rmc_main.c:530]: ok. I will fix it

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Pull request merged**